src/Api/RomeoShop.Backend.Api/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using RomeoShop.Backend.Contracts.Catalog;
+using RomeoShop.Backend.Cqrs.Commands.Product;
+using RomeoShop.Backend.Cqrs.Queries.Product;
+using RomeoShop.Backend.Dtos.Catalog;
+
+namespace RomeoShop.Backend.Api.Controllers
+{
+    [Route("api/v1/catalog")]
+    [ApiController]
+    public class CatalogController : Controller
+    {
+        private readonly IMapper _mapper;
+        private readonly IMediator _mediator;
+
+        public CatalogController(IMapper mapper, IMediator mediator)
+        {
+            _mapper = mapper;
+            _mediator = mediator;
+        }
+
+        [HttpGet]
+        [Route("products")]
+        public async Task<IActionResult> GetAllProducts()
+        {
+            var query = new GetAllProductsQuery();
+            var dtos = await _mediator.Send(query);
+            var result = _mapper.Map<IReadOnlyCollection<ProductResponseContract>>(dtos);
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("product/{id:int}")]
+        public async Task<IActionResult> GetProduct(int id)
+        {
+            var query = new GetProductQuery(id);
+            var dto = await _mediator.Send(query);
+            var result = _mapper.Map<ProductResponseContract>(dto);
+            return Ok(result);
+        }
+
+        [HttpDelete]
+        [Route("product/{id:int}")]
+        public async Task<IActionResult> DeleteProduct(int id)
+        {
+            var command = new DeleteProductCommand(id);
+            var result = await _mediator.Send(command);
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> PostProduct([FromBody] NewProductRequestContract request)
+        {
+            var dto = _mapper.Map<NewProductDto>(request);
+            var command = new PostProductCommand(dto);
+            var result = await _mediator.Send(command);
+            return Ok(result);
```

Comment:
Касается всех методов где тебе ничего не нужно возвращать кроме статуса 200. Нет смысла возвращать _Unit_

```suggestion
            await _mediator.Send(command);
            return Ok();
```